sunpy/map/mapbase.py

```diff
@@ -679,8 +679,21 @@ def spatial_units(self):
         """
         Image coordinate units along the x and y axes (i.e. cunit1, cunit2).
         """
-        return SpatialPair(u.Unit(self.meta.get('cunit1', 'arcsec')),
-                           u.Unit(self.meta.get('cunit2', 'arcsec')))
+        cunit1 = self.meta.get('cunit1')
+        cunit2 = self.meta.get('cunit2')
+
+        msg = ('Image coordinate units for axis {} not present in metadata. '
```

Comment:
I just remembered that we have `GenericMap._validate_meta` as a place for doing these checks. Which causes the errors to be raised on instantiation, which is probably preferable than at a random point later.

Reply:
I think doing this will completely prevent creation of the `Map` object, and therefore not give the user a way to manually specify the units before doing other operations on the map.

Reply:
What do we want? allow the creation of a `Map` raising this error when it's used by:
- `plot`, 
- `superpixel`
- `draw_rectangle`
- `shift`
- `_reference_longitdue`, `_reference_latitude`
- `get_lon_lat`
   - `world_to_pixel`, 
   - `resample`
   - `rotate` 
   - `submap`
- `wcs`
  - `coordinate_frame`
  - `as_mpl_axes` 

Or tell the user at creation: Hey!! your map doesn't has enough information to allow you to play with it correctly.
I just imagine that could be really annoying to get this type of message after I've done some analysis (don't know which) to then been told that my map is not "good". What's worse... I could have been saving these maps which in the future I want to plot, to discover that I have to reprocess all of them (imagine 1000s).

From my point of view, I think we should [fail fast](https://en.wikipedia.org/wiki/Fail-fast).

Reply:
The use case I was thinking of was using `sunpy.map.Map` to load a `.fits` file, and extract the data.  For this the coordinate system doesn't have to be defined, but `sunpy.map.Map` needs to be able to make a map that doesn't have enough metadata to do any of the plotting/coordinate stuff. Unless there's another way to load data from a `.fits` file without going through `Map`?

Reply:
Looks like `sunpy.io` stuff can be used, so I'd be 👍 on failing fast.

Reply:
I think we should couple this validation with the discussion starting in #1832 about formalising the minimum metadata needed to build a map. Then we should document all of this as well.

@dstansby will you be able to shuffle this around into the validation method and then we can merge this and continue from there?